CompilerSource/parsing/parser.cpp

```diff
@@ -990,7 +987,11 @@ void AstBuilder::TryParseTypeSpecifier(FullType *type) {
         if (contains_decflag_bitmask(type->flags, "unsigned") && token.type == TT_SIGNED) {
           herr->Error(token) << "Conflicting use of 'unsigned' and 'signed' in the same type specifier";
         } else if (contains_decflag_bitmask(type->flags, token.content)) {
-          herr->Warning(token) << "Duplicate usage of flags in type specifier";
+          if (first_signed && token.type == TT_SIGNED) {
```

Comment:
The `first_signed` bool is used to accept declarations like `signed int x`, because the `signed flag` is the default. Without this flag, encountering the `signed` token would be considered a duplicate usage of flags.

Reply:
No need to handle that here. It's not *syntactically* invalid to say `signed signed long unsigned signed int;`—the parser can *still build a tree.* We need to check that in a separate semantic checking layer afterward. Otherwise it creates a lot of work and awkward code flow here.

Another thing to check for in this layer: Does the "local" or "global" keyword appear in a type cast?

Reply:
Okay, I removed them. Regarding local and global, I found that they need some extra work beyond just the type cast check, so I think it would be better to address them in the upcoming PR.